File: harness/lua/tests/test_environment_isolation.lua
local t = require("support.assertions")

for _, name in ipairs({
    "LUA_INIT",
    "LUA_INIT_5_5",
    "LUA_PATH",
    "LUA_PATH_5_5",
    "LUA_CPATH",
    "LUA_CPATH_5_5",
}) do
    t.equal(os.getenv(name), nil)
end

local home = assert(os.getenv("HOME"))
local temporary = assert(os.getenv("POLYGLOT_LUA_TEST_TMP"))
t.truth(home:match("/home$") ~= nil)
t.same(temporary, assert(os.getenv("TMPDIR")))
t.matches(package.path, "harness/lua")
t.falsey(package.path:match("%.luarocks"))

-- 每个文件运行在新进程中；真实用户启动脚本、module path 和持久目录都不可见。
t.done()

File: harness/ruby/tests/test_environment_isolation.rb
# frozen_string_literal: true

require "assertions"
require "pathname"
require "rubygems"

A = PolyglotAssertions

home = Pathname.new(ENV.fetch("HOME"))
gem_home = Pathname.new(ENV.fetch("GEM_HOME"))
bundle_home = Pathname.new(ENV.fetch("BUNDLE_USER_HOME"))
A.matches(%r{\A/tmp/polyglot-ruby-harness/}, home.to_s)
A.truth(gem_home.to_s.start_with?(home.parent.to_s))
A.truth(bundle_home.to_s.start_with?(home.parent.to_s))
A.equal(gem_home.to_s, Gem.dir)
A.includes(Gem.path, gem_home.to_s)
A.equal("/dev/null", ENV.fetch("GEMRC"))
A.equal("true", ENV.fetch("BUNDLE_ALLOW_OFFLINE_INSTALL"))
A.equal("UTF-8", Encoding.default_external.name)
A.equal("UTC", ENV.fetch("TZ"))
A.equal(0, Time.local(2026, 1, 1).utc_offset)

A.done
